### PR TITLE
Issue 761  nlm

### PR DESCRIPTION
**Pull request by adler-j**
New PR from my account. Lets get this out of the way.

### CODE REVIEW

**Comment by adler-j** [comment]
Did some cleanup, should be reviewable. Keep in mind this in contrib stuff

File: odl/contrib/solvers/examples/tomography_nonlocalmeans.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (kohr-h) [inline odl/contrib/solvers/examples/tomography_nonlocalmeans.py]: Remove header

File: odl/contrib/solvers/functional/nonlocalmeans_functionals.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (kohr-h) [inline odl/contrib/solvers/functional/nonlocalmeans_functionals.py]: Update

File: odl/contrib/solvers/functional/nonlocalmeans_functionals.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        \\int_\\Omega
+        e^{-\\frac{(G_a * |u(x + \cdot) - u(y + \cdot)|^2)(0)}{h^2}}
+        dy
+
+    See FlexISP_
+
+    References
+    ----------
+    .. _Buades2005: http://ieeexplore.ieee.org/document/1467423/
+    .. _FlexISP: http://dl.acm.org/citation.cfm?doid=2661229.2661260
```
Comment (kohr-h) [inline odl/contrib/solvers/functional/nonlocalmeans_functionals.py]: Reference style

File: odl/contrib/solvers/functional/nonlocalmeans_functionals.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    :math:`h` is a parameter and :math:`*` denotes convolution and :math:`C(x)`
+    is a normalization constant
+
+    .. math::
+        C(x) =
+        \\int_\\Omega
+        e^{-\\frac{(G_a * |u(x + \cdot) - u(y + \cdot)|^2)(0)}{h^2}}
+        dy
+
+    See FlexISP_
```
Comment (kohr-h) [inline odl/contrib/solvers/functional/nonlocalmeans_functionals.py]: For the prox?

File: odl/contrib/solvers/functional/nonlocalmeans_functionals.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        self.impl = impl
+        self.patch_size = patch_size
+        self.patch_distance = patch_distance
+        super().__init__(space=space, linear=False, grad_lipschitz=np.nan)
+
+    @property
+    def proximal(self):
+        func = self
+
+        class NLMProximal(Operator):
```
Comment (kohr-h) [inline odl/contrib/solvers/functional/nonlocalmeans_functionals.py]: Expression for this guy?

File: odl/contrib/solvers/README.md [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# odl.contrib.solvers
+
+- Contributed additions to the [odl.solvers](../../solvers) package.
+- Contains application specific functionals and solvers
```
Comment (kohr-h) [inline odl/contrib/solvers/README.md]: [...] package, intended for application-specific functionals and solvers.

**Pull request merged**